src/iosurface.cs

```diff
@@ -0,0 +1,239 @@
+		NSString PlaneBaseKey { get; }
+		
+		[Field ("IOSurfacePropertyKeyPlaneBytesPerElement")]
+		[iOS (10,0), Mac (10,12)]
+		NSString PlaneBytesPerElementKey { get; }
+		
+		[Field ("IOSurfacePropertyKeyPlaneElementWidth")]
+		[iOS (10,0), Mac (10,12)]
+		NSString PlaneElementWidthKey { get; }
+		
+		[Field ("IOSurfacePropertyKeyPlaneElementHeight")]
+		[iOS (10,0), Mac (10,12)]
+		NSString PlaneElementHeightKey { get; }
+		
+		[Field ("IOSurfacePropertyKeyCacheMode")]
+		[iOS (10,0), Mac (10,12)]
+		NSString CacheModeKey { get; }
+		
+		[Field ("IOSurfacePropertyKeyPixelFormat")]
+		[iOS (10,0), Mac (10,12)]
+		NSString PixelFormatKey { get; }
+		
+		[Field ("IOSurfacePropertyKeyPixelSizeCastingAllowed")]
+		[iOS (10,0), Mac (10,12)]
+		NSString PixelSizeCastingAllowedKey { get; } 
+	}
+
+	[StrongDictionary ("IOSurfacePropertyKey")]
+	partial interface IOSurfaceOptions {
+		nint AllocSize { get; set; }
+		nint Width { get; set; }
+		nint Height { get; set; }
+		nint BytesPerRow { get; set; }
+		nint BytesPerElement { get; set; }
+		nint ElementWidth { get; set; }
+		nint ElementHeight { get; set; }
+		nint Offset { get; set; }
+		NSDictionary [] PlaneInfo { get; set; }
+		nint PlaneWidth { get; set; }
+		nint PlaneHeight { get; set; }
+		nint PlaneBytesPerRow { get; set; }
+		nint PlaneOffset { get; set; }
+		nint PlaneSize { get; set; }
+		nint PlaneBase { get; set; }
+		nint PlaneBytesPerElement { get; set; }
+		nint PlaneElementWidth { get; set; }
+		nint PlaneElementHeight { get; set; }
+		IOSurfaceMemoryMap CacheMode { get; set; }
+		uint PixelFormat { get; set; }
+		bool PixelSizeCastingAllowed { get; set; }
+	}
+
+	// @interface IOSurface : NSObject <NSSecureCoding>
+	[iOS (10,0), Mac(10,12)]
+	[BaseType (typeof(NSObject))]
+	interface IOSurface : NSSecureCoding
+	{
+		[Internal, Export ("initWithProperties:")]
+		IntPtr Constructor (NSDictionary properties);
+
+		[Wrap ("this (properties == null ? null : properties.Dictionary)")]
+		IntPtr Constructor (IOSurfaceOptions properties);
+	
+		[Internal, Export ("lockWithOptions:seed:")]
+		int _Lock (IOSurfaceLockOptions options, IntPtr seedPtr);
+	
+		[Internal, Export ("unlockWithOptions:seed:")]
+		int _Unlock (IOSurfaceLockOptions options, IntPtr seed);
+	
+		[Export ("allocationSize")]
+		nint AllocationSize { get; }
+	
+		[Export ("width")]
+		nint Width { get; }
+	
+		[Export ("height")]
+		nint Height { get; }
+	
+		[Export ("baseAddress")]
+		IntPtr BaseAddress { get; }
+	
+		[Export ("pixelFormat")]
+		uint PixelFormat { get; }
+	
+		[Export ("bytesPerRow")]
+		nint BytesPerRow { get; }
+	
+		[Export ("bytesPerElement")]
+		nint BytesPerElement { get; }
+	
+		[Export ("elementWidth")]
+		nint ElementWidth { get; }
+	
+		[Export ("elementHeight")]
+		nint ElementHeight { get; }
+	
+		[Export ("seed")]
+		uint Seed { get; }
+	
+		[Export ("planeCount")]
+		nuint PlaneCount { get; }
+	
+		[Export ("widthOfPlaneAtIndex:")]
+		nint WidthOfPlane (nuint planeIndex);
```

Comment:
Also not sure `OfPlane` is really needed (implied by the parameter name), by Swift kept it around...